src/ivectorbin/compute-vad-from-frame-likes.cc

```diff
@@ -0,0 +1,212 @@
+}
+
+}
+
+int main(int argc, char *argv[]) {
+  using namespace kaldi;
+  typedef kaldi::int32 int32;
+  try {
+    const char *usage =
+      "This program computes frame-level voice activity decisions from a\n"
+      "set of input frame-level log-likelihoods.  Frames are assigned\n"
+      "labels according to the class for which the log-likelihood\n"
+      "(optionally weighted by a prior) is maximal.  The class labels are\n"
+      "determined by the order of inputs on the command line.  See options\n"
+      "for more details.\n"
+      "See also: fgmm-global-get-frame-likes, compute-vad, merge-vads\n"
+      "\n"
+      "Usage: compute-vad-from-frame-likes [options] <likes-rspecifier-1>\n"
+      "    ... <likes-rspecifier-n> <vad-wspecifier>\n"
+      "e.g.: compute-vad-from-frame-likes --map=label_map.txt\n"
+      "    scp:likes1.scp scp:likes2.scp ark:vad.ark\n";
+
+    ParseOptions po(usage);
+    std::string map_rxfilename;
+    std::string priors_str;
+
+    po.Register("map", &map_rxfilename, "Table that defines the frame-level"
+    " labels.  For each row, the first field is the zero-based index of the"
+    " input likelihood archive and the second field is the associated"
+    " integer label.");
+
+    po.Register("priors", &priors_str, "Comma-separated list that specifies"
+    " the priors for each class.  The order of the floats corresponds to the"
+    " index of the input archives.  E.g., --priors=0.5,0.2,0.3");
+
+    po.Read(argc, argv);
+    if (po.NumArgs() < 3) {
+      po.PrintUsage();
+      exit(1);
+    }
+
+    unordered_map<int32, int32> map;
+    std::vector<BaseFloat> priors;
+    int32 num_classes = po.NumArgs() - 1;
+    PrepareMap(map_rxfilename, num_classes, &map);
+    PreparePriors(priors_str, num_classes, &priors);
+
+    SequentialBaseFloatVectorReader first_reader(po.GetArg(1));
+    std::vector<RandomAccessBaseFloatVectorReader *> readers;
+    std::string vad_wspecifier = po.GetArg(po.NumArgs());
+    BaseFloatVectorWriter vad_writer(vad_wspecifier);
+
+    for (int32 i = 2; i < po.NumArgs(); i++) {
+      RandomAccessBaseFloatVectorReader *reader
+        = new RandomAccessBaseFloatVectorReader(po.GetArg(i));
+      readers.push_back(reader);
+    }
+
+    int32 num_done = 0, num_err = 0;
+    for (;!first_reader.Done(); first_reader.Next()) {
+      std::string utt = first_reader.Key();
+      Vector<BaseFloat> like(first_reader.Value());
+      int32 like_dim = like.Dim();
+      std::vector<Vector<BaseFloat> > likes;
+      likes.push_back(like);
+      if (like_dim == 0) {
+        KALDI_WARN << "Empty vector for utterance " << utt;
+        num_err++;
+        continue;
+      }
+      for (int32 i = 0; i < num_classes - 1; i++) {
+        if (!readers[i]->HasKey(utt)) {
+          KALDI_WARN << "No vector for utterance " << utt;
+          num_err++;
+          continue;
+        }
+        Vector<BaseFloat> other_like(readers[i]->Value(utt));
+        if (like_dim != other_like.Dim()) {
```

Comment:
Generally we'd print a warning and increase num_err if something like this happens.